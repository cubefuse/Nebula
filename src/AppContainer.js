// @flow
import { connect } from "react-redux";
import App from "./App";

const mapStateToProps = state => {
  return { ipfs: state.ipfs };
};

const AppContainer = connect(mapStateToProps)(App);
export default AppContainer;
