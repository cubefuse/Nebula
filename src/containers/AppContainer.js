import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import App from "../components/App";
import { createIpfsInstance, getIpfsNodeStats } from "../redux/ipfs.redux";

function mapStateToProps(state) {
  return { ipfs: state.ipfs };
}

function mapDispatchToProps(dispatch) {
  return {
    createIpfsInstance: () => {
      dispatch(createIpfsInstance());
    },
    getIpfsNodeStats: () => {
      dispatch(getIpfsNodeStats());
    }
  };
}

const AppContainer = withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(App)
);
export default AppContainer;
