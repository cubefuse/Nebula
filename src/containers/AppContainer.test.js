import React from "react";
import "jest-dom/extend-expect";
import TestHelpers from "../util/test-helpers";
import AppContainer from "./AppContainer";

test("can render with initial state", () => {
  const node = TestHelpers.renderWithRouterAndRedux(<AppContainer />);
  expect(node).toMatchSnapshot();
});
