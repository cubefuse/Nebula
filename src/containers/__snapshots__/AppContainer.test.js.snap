// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with initial state 1`] = `
Object {
  "container": <div>
    <div
      class="ant-layout"
      style="min-height: 100vh;"
    >
      <div
        class="header ant-layout-header"
      >
        <div
          style="color: white;"
        >
          <span
            style="font-size: x-large; padding-right: 5px;"
          >
            <strong>
              Nebula
            </strong>
          </span>
          <span
            style="font-size: large; padding-right: 10px;"
          >
            Web UI
          </span>
          <span
            style="font-size: small;"
          >
            POC-01
          </span>
        </div>
      </div>
      <div
        class="ant-layout-content"
        style="padding: 0px 24px;"
      >
        <div
          class="ant-breadcrumb"
          style="margin: 16px 0px;"
        >
          <span>
            <span
              class="ant-breadcrumb-link"
            >
              <a
                href="#/"
              >
                Home
              </a>
            </span>
            <span
              class="ant-breadcrumb-separator"
            >
              /
            </span>
          </span>
        </div>
        <div
          class="ant-layout"
          style="padding: 24px 0px; background: rgb(255, 255, 255); height: 100%;"
        >
          <div
            class="ant-layout-content"
            style="text-align: center;"
          >
            <div
              class="ant-spin ant-spin-lg ant-spin-spinning"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i />
                <i />
                <i />
                <i />
              </span>
            </div>
            <h1
              style="font-size: xx-large; color: gray;"
            >
              Connecting to IPFS Network
            </h1>
          </div>
        </div>
      </div>
      <div
        class="ant-layout-footer"
        style="text-align: center;"
      >
        Copyright Â© 2018 Cubefuse. Distributed under MIT license.
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
